apps/api/src/campaign-application/campaign-application.service.ts

```diff
@@ -1,26 +1,75 @@
-import { BadRequestException, HttpException, HttpStatus, Injectable } from '@nestjs/common'
+import {
+  BadRequestException,
+  HttpException,
+  HttpStatus,
+  Injectable,
+  Logger,
+  NotFoundException,
+} from '@nestjs/common'
 import { CreateCampaignApplicationDto } from './dto/create-campaign-application.dto'
 import { UpdateCampaignApplicationDto } from './dto/update-campaign-application.dto'
+import { PrismaService } from '../prisma/prisma.service'
+import { OrganizerService } from '../organizer/organizer.service'
+import { Person } from '@prisma/client'
 
 @Injectable()
 export class CampaignApplicationService {
+  constructor(private prisma: PrismaService, private organizerService: OrganizerService) {}
   async getCampaignByIdWithPersonIds(id: string): Promise<UpdateCampaignApplicationDto> {
     throw new Error('Method not implemented.')
   }
 
-  create(createCampaignApplicationDto: CreateCampaignApplicationDto) {
-    if (
-      !createCampaignApplicationDto.acceptTermsAndConditions ||
-      !createCampaignApplicationDto.transparencyTermsAccepted ||
-      !createCampaignApplicationDto.personalInformationProcessingAccepted
-    ) {
-      throw new BadRequestException('All agreements must be checked')
+  async create(createCampaignApplicationDto: CreateCampaignApplicationDto, person: Person) {
+    try {
+      if (
+        createCampaignApplicationDto.acceptTermsAndConditions === false ||
+        createCampaignApplicationDto.transparencyTermsAccepted === false ||
+        createCampaignApplicationDto.personalInformationProcessingAccepted === false
+      ) {
+        throw new BadRequestException('All agreements must be checked')
+      }
+
+      let organizer = await this.prisma.organizer.findUnique({
+        where: { personId: person.id },
+      })
+
+      if (!organizer) {
+        organizer = await this.organizerService.create({
+          personId: person.id,
+        })
+      }
```

Comment:
Not sure if we should create a person here. I'd say - if the organizer is missing - no new campaignApplication.

Reply:
Here I am not creating person. Initialy there are no organizers, so when a person creates a new campaignApplication he becomes a organizer => creating organizer with the person's id